### ISSUE TITLE
Does it work under windows

### ISSUE THREAD

**Issue by TkonstAL** [CompVis/stable-diffusion#37]
Good afternoon! I tried to put it under windows 10 but I couldn't. Anyone got something like this?

**Comment by irmrenders** [comment]
It works on windows 10

**Comment by breadbrowser** [comment]
just use this https://huggingface.co/spaces/stabilityai/stable-diffusion

**Comment by smoran** [comment]
Works locally for me on a Win10 1080ti (11Gb RAM).
I can run 512x512 1 sample.
512x768 one sample with model.half()
And can run 786x1024 with the optimized fork.

**Comment by Vanceagher** [comment]
@smoran what optimized for are you refering to?

**Comment by Louis24** [comment]
> It works on windows 10

hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

shall i change the num_worker=0?

**Comment by smoran** [comment]
> @smoran what optimized for are you refering to?

Sorry for the late reply. I didn't get notified about your mention.
This version:
https://github.com/basujindal/stable-diffusion
Allows using SD with 3.8Gb for a 512x512.
Please note it goes on the expense of run time

**Comment by smoran** [comment]
> > It works on windows 10
> 
> hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> 
> shall i change the num_worker=0?

Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function.
This will save you a couple of GB. 
If you still get CUDA OOM,  try running a lower resolution:
Do 448x448 (--H 448 --W 448). It will save you a lot of GPU.
You can also use this version:  https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower)
Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)

**Comment by Louis24** [comment]
> > > It works on windows 10
> > 
> > 
> > hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> > shall i change the num_worker=0?
> 
> Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function. This will save you a couple of GB. If you still get CUDA OOM, try running a lower resolution: Do 448x448 (--H 448 --W 448). It will save you a lot of GPU. You can also use this version: https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower) Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)

Yep it works thx, may I know the memory and flops reduces? is the all goes to 1/2?

**Comment by Louis24** [comment]
> > > > It works on windows 10
> > > 
> > > 
> > > hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> > > shall i change the num_worker=0?
> > 
> > 
> > Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function. This will save you a couple of GB. If you still get CUDA OOM, try running a lower resolution: Do 448x448 (--H 448 --W 448). It will save you a lot of GPU. You can also use this version: https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower) Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)
> 
> Yep it works thx, may I know the memory and flops reduces? is the all goes to 1/2?

well I think should be decrese more than 1/2 cause I have tried the size 128*128 it can not word until i change .half()

**Comment by smoran** [comment]
> > > > > It works on windows 10
> > > > 
> > > > 
> > > > hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> > > > shall i change the num_worker=0?
> > > 
> > > 
> > > Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function. This will save you a couple of GB. If you still get CUDA OOM, try running a lower resolution: Do 448x448 (--H 448 --W 448). It will save you a lot of GPU. You can also use this version: https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower) Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)
> > 
> > 
> > Yep it works thx, may I know the memory and flops reduces? is the all goes to 1/2?
> 
> well I think should be decrese more than 1/2 cause I have tried the size 128*128 it can not word until i change .half()



> > > > It works on windows 10
> > > 
> > > 
> > > hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> > > shall i change the num_worker=0?
> > 
> > 
> > Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function. This will save you a couple of GB. If you still get CUDA OOM, try running a lower resolution: Do 448x448 (--H 448 --W 448). It will save you a lot of GPU. You can also use this version: https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower) Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)
> 
> Yep it works thx, may I know the memory and flops reduces? is the all goes to 1/2?

It changes the precision of the model from 32bit to 64bit.
Not sure about the exact memory saving, but it is indeed around half.
Note that all models should be halved in case more than one is being used.

**Comment by smoran** [comment]
> > > > > It works on windows 10
> > > > 
> > > > 
> > > > hi I got this error RuntimeError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 8.00 GiB total capacity; 5.37 GiB already allocated; 0 bytes free; 5.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> > > > shall i change the num_worker=0?
> > > 
> > > 
> > > Not the wrokers. You can Either half the model (Add model.half in the code after the load_model function. This will save you a couple of GB. If you still get CUDA OOM, try running a lower resolution: Do 448x448 (--H 448 --W 448). It will save you a lot of GPU. You can also use this version: https://github.com/basujindal/stable-diffusion Which dramatically lowers the GPU Usage (but runs slower) Running on a lower res, you can then use a upsclaer like RealESGAN to upper the resolution to up to x4 (it doesn't cost a lot of GPU)
> > 
> > 
> > Yep it works thx, may I know the memory and flops reduces? is the all goes to 1/2?
> 
> well I think should be decrese more than 1/2 cause I have tried the size 128*128 it can not word until i change .half()

The model was trained for 512x512. When using much lower resolutions, it will note be able to generate proper diffusions.
the optimal res is 512x512. too hight/lower than this, might cause artifacts/issues. 
IThe best thing is to maintain 512 on width or height and than upscale using other models (RealESGAN does an amazing work [including face enhancements])
